openai-fine-tuning/OpenAI_fine-tuning.ipynb

```diff
@@ -0,0 +1,1230 @@
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Number of examples in master_qa_list: \", len(master_qa_list))\n",
+    "# Example: Print the first 10 examples\n",
+    "for data in master_qa_list[:10]:\n",
+    "    print(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Review the QA pairs in Label Studio\n",
+    "We can now review the QA pairs in Label Studio to ensure they are of high quality."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def format_qa_list_for_label_studio(qa_list):\n",
+    "    \"\"\"\n",
+    "    Format the Q&A list into the desired structure for Label Studio.\n",
+    "    \n",
+    "    :param qa_list: List of dictionaries containing 'prompt' and 'completion' keys\n",
+    "    :return: Formatted list of Q&A pairs\n",
+    "    \"\"\"\n",
+    "    formatted_list = []\n",
+    "    for qa_pair in qa_list:\n",
+    "        formatted_qa = {\n",
+    "            \"data\": {\n",
+    "                \"question\": qa_pair['prompt'],\n",
+    "                \"article_title\": qa_pair['article_title'],\n",
+    "                \"context\": qa_pair['new_content']\n",
+    "            },\n",
+    "            \"predictions\": [\n",
+    "                {\n",
+    "                    \"model_version\": \"v1\",\n",
+    "                    \"result\": [\n",
+    "                        {\n",
+    "                            \"from_name\": \"completion\",\n",
+    "                            \"to_name\": \"question\",\n",
+    "                            \"type\": \"textarea\",\n",
+    "                            \"value\": {\n",
+    "                                \"text\": [qa_pair['completion']]\n",
+    "                            }\n",
+    "                        }\n",
+    "                    ]\n",
+    "                }\n",
+    "            ]\n",
+    "        }\n",
+    "        formatted_list.append(formatted_qa)\n",
+    "    return formatted_list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 59,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Save the formatted version of master_qa_list\n",
+    "formatted_qa_list = format_qa_list_for_label_studio(master_qa_list)\n",
+    "with open('formatted_qa_list.json', 'w') as f:\n",
+    "    json.dump(formatted_qa_list, f, indent=2)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 60,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the URL where Label Studio is accessible and the API key for your user account\n",
+    "LABEL_STUDIO_URL = 'http://localhost:8080'  # Your Label Studio instance URL\n",
+    "API_KEY = 'b7fc43a9abe38ddbebca580cd1a6fd03b3778e8a'  # Your Label Studio API key\n",
```

Comment:
Probably you should leave the API key blank so we don't post your API key to the internet (you might want to cycle it now!)

Reply:
Good catch. It shouldn't matter since I'm running it on localhost, but I removed it just in case.